### ISSUE TITLE
library_annotations is not triggered for @Deprecated annotation

### ISSUE THREAD

**Issue by alexeyinkin** [dart-lang/linter#4026]
**Describe the issue**
`library_annotations` is not triggered for `@Deprecated` annotation.

**To Reproduce**

/lib/library.dart:
```dart
@Deprecated('')
import 'package:test/test.dart'; // ignore: unused_import
```

/lib/main.dart:
```dart
import 'library.dart'; // ignore: unused_import
```

/analysis_options.yaml:
```yaml
linter:
  rules:
    - library_annotations
```

/pubspec.yaml:
```dart
name: my_project
environment:
  sdk: '>=2.19.0 <3.0.0'

dependencies:
  test:
```

**Expected behavior**
`dart analyze` should produce `library_annotations` linter issue because `@Deprecated` is applied to `import` rather than `library` directive in addition to `deprecated_member_use_from_same_package`.

**Actual behavior**
```dart
   info • lib/main.dart:1:1 • 'package:my_project/library.dart' is
          deprecated and shouldn't be used. Try replacing the use of the
          deprecated member with the replacement. •
          deprecated_member_use_from_same_package

1 issue found.
```

**Additional context**
Just to make sure that everything is configured correctly:
1. Replace `@Deprecated('')` annotation with `@TestOn('browser')` as in the lint example: https://dart-lang.github.io/linter/lints/library_annotations.html

```dart
@TestOn('browser')
import 'package:test/test.dart';
```
2. Witness the lint triggered:
```
   info • lib/library.dart:1:1 • This annotation must be attached to a
          library directive. Try attaching library annotations to library
          directives. • library_annotations

1 issue found.
```

This difference in handling `@Deprecated` and `@TestOn` suggests that `@Deprecated` was applied to the `import` and not the entire library. But in that case `deprecated_member_use_from_same_package` should not have been triggered in `main.dart` when importing `library.dart` (apart from the obvious question what does a deprecated import mean).

**Comment by lrhn** [comment]
A deprecated an import has no meaning.

Deprecation of a declaration signifies that the member will go away, become unavailable, at some point in the future.
Applying that to an import is not useful, since it only affects the library where the import occurs, and the library author is entirely in control of whether to remove that import or not. It's like deprecating a private member - it might make a kind of sense, but it's not actually useful.

Applying deprecation to an export is useful, because it means that some declarations in the library's export scope will go away, which affects other libraries. (I don't think we support deprecating exports yet, but it would make sense.)

So assuming that a `@Deprecated` on the first import really should have been on the library, seems warranted.
